cni/telemetry/client/telemetry_client.go

```diff
@@ -0,0 +1,118 @@
+package telemetryclient
+
+import (
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/Azure/azure-container-networking/telemetry"
+	"go.uber.org/zap"
+)
+
+const (
+	telemetryNumberRetries          = 5
+	telemetryWaitTimeInMilliseconds = 200
+)
+
+type TelemetryClient struct {
+	cniReportSettings *telemetry.CNIReport
+	tb                *telemetry.TelemetryBuffer
+	logger            *zap.Logger
+	lock              sync.Mutex
```

Comment:
is CNI spinning goroutines? What is this lock serializing?

Reply:
currently we don't send telemetry from multiple goroutines as far as I can tell, but if we want to send telemetry from anywhere in the codebase in the future, I would prefer to serialize this and not worry about races etc. later, but I can remove the lock for now if you would like.